README.md

```diff
@@ -0,0 +1,315 @@
+    // that package path to the list we defined earlier.
+    for _, name := range keys {
+        // We don't want Aws prefix in all kinds.
+        kind := strings.TrimPrefix(strcase.ToCamel(name), "Aws")
+        if err := crdGen.Generate(version, kind, resources[name]); err != nil {
+            panic(errors.Wrap(err, "cannot generate crd"))
+        }
+        if err := tfGen.Generate(version, kind, name, "id"); err != nil {
+            panic(errors.Wrap(err, "cannot generate terraformed"))
+        }
+        ctrlPkgPath, err := ctrlGen.Generate(versionGen.Package().Path(), kind)
+        if err != nil {
+            panic(errors.Wrap(err, "cannot generate controller"))
+        }
+        // Here we add the version so that the `zz_setup.go` that is called in
+        // `main.go` can include it.
+        controllerPkgList = append(controllerPkgList, ctrlPkgPath)
+    }
+    // Here we add the version so that the `zz_register.go` can include it.
+    versionPkgList = append(versionPkgList, versionGen.Package().Path())
+}
+
+// Since the list of versions is now complete, we can write the `apis/zz_register.go`
+// file.
+if err := pipeline.NewRegisterGenerator(wd, modulePath).Generate(versionPkgList); err != nil {
+    panic(errors.Wrap(err, "cannot generate register file"))
+}
+
+// Since the list of controllers is now complete, we can write the
+// `internal/controller/zz_setup.go` file.
+if err := pipeline.NewSetupGenerator(wd, modulePath).Generate(controllerPkgList); err != nil {
+    panic(errors.Wrap(err, "cannot generate setup file"))
+}
+
+// We run goimports on generated files for easier human interaction.
+if err := exec.Command("bash", "-c", "goimports -w $(find apis -iname 'zz_*')").Run(); err != nil {
+    panic(errors.Wrap(err, "cannot run goimports for apis folder"))
+}
+if err := exec.Command("bash", "-c", "goimports -w $(find internal -iname 'zz_*')").Run(); err != nil {
+    panic(errors.Wrap(err, "cannot run goimports for internal folder"))
+}
+```
+
+This code excerpt is the main piece that you need to implement to get the code
+generators working. After the first run, you'll see many CRDs generated. In order
+to run other generic code generators, run the following command:
+```
+make generate
+```
+
+### Provider Config Builder
+
+<to be filled>
+
+### Local Testing
+
+Once you complete the first section, you'll have many CRDs in `package/crds`. You
+can install them into your cluster by running `kubectl apply -f package/crds` and
+then start `cmd/provider/main.go` to start their controllers.
+
+At this point, the managed resources you create will be reconciled.
+
+### Packaging
+
+> This section might go away once we have a provider-tf-template or a branch
+> in provider-template that you can use as template for your Terraform-based
+> provider.
+
+Since the common controller shells out to Terraform CLI, which in turn talks to
+its provider plugin, we'd like to make those two available in the final controller
```

Comment:
Non-blocking todo for later: We should have some kind of architectural overview on how these components work together. Especially the new components that come with Terraform.